FABulous/FABulous_CLI/helper.py

```diff
@@ -470,3 +467,117 @@ def execute(self) -> bool:
     def get_exit_code(self) -> int:
         """Get the final exit code from pipeline execution."""
         return self.final_exit_code
+
+
+def clone_git_repo(repo_url: str, target_dir: Path, branch: str = "main") -> bool:
+    """Clone or update a GitHub repository.
+
+    Parameters
+    ----------
+        repo_url: str
+            GitHub repository URL (e.g., "https://github.com/user/repo.git")
+        target_dir: Path
+            Local directory to clone/download to
+        branch: str
+            Git branch to checkout (default: "main")
+
+    Returns
+    -------
+        True if successful, False otherwise
+
+    Raises
+    ------
+    """
+    try:
+        logger.info(f"Downloading reference projects from {repo_url}")
+
+        if target_dir.exists():
+            # If directory exists, try to update it
+            if (target_dir / ".git").exists():
+                logger.info("Updating existing repository...")
+                result = subprocess.run(
+                    ["git", "pull", "origin", branch],
+                    cwd=target_dir,
+                    capture_output=True,
+                    text=True,
+                    timeout=60,
+                )
+                if result.returncode != 0:
+                    logger.warning(f"Git pull failed: {result.stderr}")
+                    logger.info("Attempting fresh clone...")
+                    shutil.rmtree(target_dir)
+                else:
+                    logger.info("✓ Repository updated successfully")
+                    return True
+            else:
+                logger.error(
+                    f"Target directory {target_dir} exists but is not a git repository."
+                    " Please remove or specify a different directory.",
+                )
+                return False
+
+        if not target_dir.exists():
+            # Fresh clone
+            logger.info("Cloning repository...")
+            target_dir.parent.mkdir(parents=True, exist_ok=True)
+
+            result = subprocess.run(
+                [
+                    "git",
+                    "clone",
+                    "--branch",
+                    branch,
+                    "--depth",
+                    "1",
+                    repo_url,
+                    str(target_dir),
+                ],
+                capture_output=True,
+                text=True,
+                timeout=120,
+            )
+
+            if result.returncode != 0:
+                logger.error(f"Failed to clone repository: {result.stderr}")
+                return False
+
+            logger.info("✓ Repository cloned successfully")
+            return True
+
+    except subprocess.TimeoutExpired:
+        logger.error("Git operation timed out")
+        return False
+    except FileNotFoundError:
+        logger.error("Git command not found. Please install git.")
```

Comment:
This is likely raised from a subprocess error?

If we do the check in the first place, then we can skip this section.

Reply:
Let's just leave it as is.